src/mono/wasm/debugger/DebuggerTestSuite/EvaluateOnCallFrameTests.cs

```diff
@@ -1178,47 +1178,67 @@ async Task EvaluateOnCallFrameFail(string call_frame_id, params (string expressi
+                    var actualValObj = GetAndAssertObjectWithName(testRootHiddenProps, expectedListRootHiddenElements[i]["name"].Value<string>());
+                    await CheckValue(actualValObj["value"], expectedListRootHiddenElements[i]["value"], "listRootHidden");
                 }
-                foreach (var item in refArrayProp)
+
+                for (int i = 0; i < expectedArrayElements.Length; i ++)
                 {
-                    item["name"] = string.Concat("arrayRootHidden[", item["name"], "]");
-                    CheckContainsJObject(testRootHiddenProps, item, item["name"].Value<string>());
+                    var actualValObj = GetAndAssertObjectWithName(testRootHiddenProps, expectedArrayElements[i]["name"].Value<string>());
+                    await CheckValue(actualValObj["value"], expectedArrayElements[i]["value"], "x");
                 }
 
-                // valuetype/class members unique names are created by concatenation with a dot
-                foreach (var item in refStructProp)
+                foreach (var structItem in expectedStructRootHiddenElements)
                 {
-                    item["name"] = string.Concat("sampleStructRootHidden.", item["name"]);
-                    CheckContainsJObject(testRootHiddenProps, item, item["name"].Value<string>());
+                    var actualValObj = GetAndAssertObjectWithName(testRootHiddenProps, structItem["name"].Value<string>());
+                    await CheckValue(actualValObj["value"], structItem["value"], "sampleStructRootHidden");
                 }
-                foreach (var item in refClassProp)
+
+                foreach (var classItem in expectedClassRootHiddenElements)
                 {
-                    item["name"] = string.Concat("sampleClassRootHidden.", item["name"]);
-                    CheckContainsJObject(testRootHiddenProps, item, item["name"].Value<string>());
+                    var actualValObj = GetAndAssertObjectWithName(testRootHiddenProps, classItem["name"].Value<string>());
+                    await CheckValue(actualValObj["value"], classItem["value"], "sampleClassRootHidden");
                 }
+
+                JObject[] expectedTestRootHiddenProps = new[]
+                {
+                    expectedArrayElements[0],
+                    expectedArrayElements[1],
+                    expectedListRootHiddenElements[0],
+                    expectedListRootHiddenElements[1],
+                    expectedClassRootHiddenElements[0],
+                    expectedClassRootHiddenElements[1],
+                    expectedStructRootHiddenElements[0],
+                    expectedStructRootHiddenElements[1]
+                };
+                // CheckProps for JArrays care about the order and here the order differs between InlineData
+                var testRootHiddenPropsSorted = new JArray(testRootHiddenProps.OrderBy(v => (string)v["name"]));
```

Comment:
The order matters, so we shouldn't adjust the results to match the test. The test types seem to have slightly different ordering of the fields/properties, which is causing the difference that you see. Making that consistent would allow not having to sort here.

I made these changes to the tests https://gist.github.com/radical/208598ae72b6310224dc531e3b66eff8 . Removing the sorting actually reveals that in case of root hidden, the members don't get added in the correct order. For example, you get `arrayRootHidden[0]` as the first member, but `arrayRootHidden[1]` is not the next one (it was the last in my run).

In general, we want to test the result as-is, as much as possible. And want to avoid adjusting the tests to match incorrect behavior. In case of incorrect behavior, we can have correct tests, but disabled with a corresponding issue.

But since we already have overlapping issues, you can either disable some tests, or patch them up as needed, to merge this PR. And in follow ups fix things to do it correctly.. thank you for being patient with it ;) Make sure to have some issue(s) tracking the TODOs.